space_packet_parser/xtce/containers.py

```diff
@@ -111,15 +120,24 @@ def from_xml(
 
             elif entry.tag == '{{{xtce}}}ContainerRefEntry'.format(
                     **ns):
-                nested_container = cls._find_container_by_name(
-                    tree,
-                    name=entry.attrib['containerRef'],
-                    ns=ns
-                )
-                entry_list.append(
-                    cls.from_xml(
-                        nested_container, ns=ns, tree=tree, parameter_lookup=parameter_lookup
+
+                if entry.attrib['containerRef'] in container_lookup:
+                    nested_container = container_lookup[entry.attrib['containerRef']]
+                else:
+                    nested_container_element = cls._find_container_by_name(
+                        tree,
+                        name=entry.attrib['containerRef'],
+                        ns=ns
                     )
+                    nested_container = cls.from_xml(
+                            nested_container_element,
+                            tree=tree,
+                            parameter_lookup=parameter_lookup,
+                            container_lookup=container_lookup,
+                            ns=ns
+                        )
```

Comment:
Is there a way to change this logic at all and put the if/else into the `_find_container_by_name`? Make that a `_get_container(name, ...)` lookup that would check the cache first and it would also create the object if it isn't found. I'm not sure if that is feasible given it is a classmethod currently and doesn't have the `self.containers` information to lookup in.
There is one other place this is used, and I wans't sure it would be helpful or not to change over to `_get_container_by_name()` so just putting this out there for your thoughts.

Reply:
I'm kind of trying to keep the finding of an Element separate from the parsing of that element. I've renamed these methods to `_get_base_container_element` and `_get_container_element` to make it clear they are just finding XML Element objects in the tree. `from_xml` is responsible for deciding if we actually want to parse those elements and store them in `container_lookup`.

There are probably more optimizations to make here but I want to move on for now.